aws/resource_aws_wafv2_rule_group_test.go

```diff
@@ -56,29 +56,34 @@ func TestAccAwsWafv2RuleGroup_Basic(t *testing.T) {
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.comparison_operator", "LT"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.field_to_match.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.field_to_match.0.query_string.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.size", "50"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.text_transformation.#", "2"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.text_transformation.2212084700.priority", "2"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.text_transformation.2212084700.type", "CMD_LINE"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.text_transformation.4292479961.priority", "5"),
-					resource.TestCheckResourceAttr(resourceName, "rule.1162901930.statement.0.size_constraint_statement.0.text_transformation.4292479961.type", "NONE"),
-					resource.TestCheckResourceAttr(resourceName, "rule.3400404505.name", "rule-1"),
-					resource.TestCheckResourceAttr(resourceName, "rule.3400404505.priority", "1"),
-					resource.TestCheckResourceAttr(resourceName, "rule.3400404505.action.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "rule.3400404505.action.0.allow.#", "0"),
-					resource.TestCheckResourceAttr(resourceName, "rule.3400404505.action.0.block.#", "0"),
-					resource.TestCheckResourceAttr(resourceName, "rule.3400404505.action.0.count.#", "1"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*", map[string]string{
+						"name":             "rule-2",
+						"priority":         "10",
+						"action.#":         "1",
+						"action.0.allow.#": "0",
+						"action.0.block.#": "1",
+						"action.0.count.#": "0",
+						"statement.#":      "1",
+						"statement.0.size_constraint_statement.#":                                 "1",
+						"statement.0.size_constraint_statement.0.comparison_operator":             "LT",
+						"statement.0.size_constraint_statement.0.field_to_match.#":                "1",
+						"statement.0.size_constraint_statement.0.field_to_match.0.query_string.#": "1",
+						"statement.0.size_constraint_statement.0.size":                            "50",
+						"statement.0.size_constraint_statement.0.text_transformation.#":           "2",
+						// TODO: super nested TypeSet check
+						"statement.0.size_constraint_statement.0.text_transformation.2212084700.priority": "2",
```

Comment:
So not a perfect setup but when dealing with a deeply nested set, validate that set independently of it's outer set

```
tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*.statement.0.size_constraint_statement.0.text_transformation.*", map[string]string{
    "priority": "2"
    // ...
},
```

Reply:
ahh yesss!

Reply:
The risk here is false positive for an inner set on "the wrong" outer set.... I have a concrete example in the AWS codebase somewhere but I wouldn't worry about it for sets with high specificity